lib/ansible/modules/files/exists.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'core'}
+
+DOCUMENTATION = r'''
+---
+module: stat
+version_added: "2.5"
+short_description: Check to see if a file or directory exists
+description:
+  - Checks to see if a file exists, similar to stat, but without calculating
+    extra checksums.
+options:
+  path:
+    description:
+      - The full path of the file to check.
+    required: true
+  follow:
+    description:
+      - Whether to follow symlinks.
+    type: bool
+    default: 'no'
+notes:
+     - For Windows targets, use the M(win_stat) module instead.
```

Comment:
This note probably no longer applies, unless you'll be making a Windows version. Since this module will not work on Windows, it should be noted. (at least I think it won't? All the windows modules use powershell so ¯\\_(ツ)_/¯

Reply:
Yah - I don't really expect to make a windows copy. I think the comment still applies though - windows users need to use win_stat if they want to check file existence. Should we reword it?

Reply:
Hrm. Yeah the comment does still apply, you're right there. A slight tweaking of the wording could be done, or not. Either way that won't block the review.